lib/rules/prop-types.js

```diff
@@ -301,6 +301,17 @@ module.exports = {
     }
 
     /**
+     * Removes quotes from around an identifier.
+     * @param {string} the identifier to strip
+     */
+    function stripQuotes(string) {
+      if (string[0] === '\'' || string[0] === '"') {
```

Comment:
I really can't think of a way to get a key name to start with a quote, without wrapping it in quotes.

I agree it feels a bit hacky. I'll work on getting a fix this weekend.